lib/docsparser.js

```diff
@@ -168,7 +178,7 @@ DocsParser.prototype.format_function = function(name, args, retval, options) {
     // return value type might be already available in some languages but
     // even then ask language specific parser if it wants it listed
     var ret_type = this.get_function_return_type(name, retval);
-    if(ret_type !== null) {
+    if(ret_type !== null && name !== 'constructor') {
```

Comment:
Not all functions named "constructor" actually have no return type. This is only the case for JS and TS and in fact `function constructor() { return 42 }` is perfectly valid JS. I would prefer it if this would be handled differently.

Reply:
Absolutely, I completely agree. In this (initial) version I tried to keep as close to the original PR as possible, although that did bug me as a "large hammer".